aws/table_aws_eks_addon_version.go

```diff
@@ -0,0 +1,122 @@
+		List: &plugin.ListConfig{
+			Hydrate: listEksAddonVersions,
+		},
+		GetMatrixItem: BuildRegionList,
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "addon_name",
+				Description: "The name of the add-on.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "addon_version",
+				Description: "The version of the add-on.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "type",
+				Description: "The type of the add-on.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "architecture",
+				Description: "The architectures that the version supports.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "compatibilities",
+				Description: "An object that represents the compatibilities of a version.",
+				Type:        proto.ColumnType_JSON,
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("AddonVersion"),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAddonVersionAkas,
+				Transform:   transform.FromValue(),
+			},
+		}),
+	}
+}
+
+type addonVersion struct {
+	AddonName       *string
+	AddonVersion    *string
+	Architecture    []*string
+	Compatibilities []*eks.Compatibility
+	Type            *string
+}
+
+//// LIST FUNCTION
+
+func listEksAddonVersions(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+	plugin.Logger(ctx).Trace("listEksAddonVersions", "AWS_REGION", region)
+
+	// Create service
+	svc, err := EksService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	err = svc.DescribeAddonVersionsPages(
+		&eks.DescribeAddonVersionsInput{},
+		func(page *eks.DescribeAddonVersionsOutput, b bool) bool {
+			for _, addon := range page.Addons {
+				for _, version := range addon.AddonVersions {
+					d.StreamListItem(ctx, addonVersion{addon.AddonName, version.AddonVersion, version.Architecture, version.Compatibilities, addon.Type})
+				}
+			}
+			return true
+		},
+	)
+
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
+
+func getAddonVersionAkas(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getAddonVersionAkas")
+	version := h.Item.(addonVersion)
+
+	commonData, err := getCommonColumns(ctx, d, h)
+	if err != nil {
+		return nil, err
+	}
+
+	commonColumnData := commonData.(*awsCommonColumnData)
+	akas := []string{"arn:" + commonColumnData.Partition + ":eks:" + commonColumnData.Region + ":" + commonColumnData.AccountId + ":addon-version/" + *version.AddonVersion}
```

Comment:
@bigdatasourav Is this an ARN format from AWS, or one we made up? If we made it up, would it be better to follow a version ARN format similar to Lambda function versions, e.g., `arn:${Partition}:lambda:${Region}:${Account}:function:${FunctionName}:${Version}`?